diff --git a/third_party/gpus/rocm/BUILD.tpl b/third_party/gpus/rocm/BUILD.tpl
index 6e05471a99..c29d37acd3 100644
--- a/third_party/gpus/rocm/BUILD.tpl
+++ b/third_party/gpus/rocm/BUILD.tpl
@@ -468,6 +468,36 @@ cc_library(
     data = [":system_libs_data"],
 )
 
+sh_binary(
+    name = "hipcc",
+    srcs = ["%{rocm_root}/bin/hipcc"],
+    visibility = ["//visibility:public"],
+)
+
+sh_binary(
+    name = "llvm-link",
+    srcs = ["%{rocm_root}/lib/llvm/bin/llvm-link"],
+    visibility = ["//visibility:public"],
+)
+
+sh_binary(
+    name = "opt",
+    srcs = ["%{rocm_root}/lib/llvm/bin/opt"],
+    visibility = ["//visibility:public"],
+)
+
+filegroup(
+    name = "ocml",
+    srcs = ["%{rocm_root}/lib/llvm/amdgcn/bitcode/ocml.bc"],
+    visibility = ["//visibility:public"],
+)
+
+filegroup(
+    name = "ockl",
+    srcs = ["%{rocm_root}/lib/llvm/amdgcn/bitcode/ockl.bc"],
+    visibility = ["//visibility:public"],
+)
+
 filegroup(
     name = "toolchain_data",
     srcs = glob(
diff --git a/third_party/rocm_device_libs/build_defs.bzl b/third_party/rocm_device_libs/build_defs.bzl
index 5b9aee4f08..f55aabe5ad 100644
--- a/third_party/rocm_device_libs/build_defs.bzl
+++ b/third_party/rocm_device_libs/build_defs.bzl
@@ -111,6 +111,7 @@ def _bitcode_library_impl(ctx):
         inputs = [strip_out],
         outputs = [final_bc],
         arguments = [strip_out.path, "-o", final_bc.path],
+        use_default_shell_env = True,
         progress_message = "Preparing final bitcode for {}".format(ctx.label.name),
         mnemonic = "BitcodeFinalize",
     )
@@ -127,17 +128,17 @@ bitcode_library = rule(
         "hdrs": attr.label_list(allow_files = [".h"]),
         "file_specific_flags": attr.string_list_dict(),
         "_clang": attr.label(
-            default = Label("@llvm-project//clang:clang"),
+            default = Label("@local_config_rocm//rocm:hipcc"),
             executable = True,
             cfg = "exec",
         ),
         "_llvm_link": attr.label(
-            default = Label("@llvm-project//llvm:llvm-link"),
+            default = Label("@local_config_rocm//rocm:llvm-link"),
             executable = True,
             cfg = "exec",
         ),
         "_opt": attr.label(
-            default = Label("@llvm-project//llvm:opt"),
+            default = Label("@local_config_rocm//rocm:opt"),
             executable = True,
             cfg = "exec",
         ),
diff --git a/third_party/rocm_device_libs/rocm_device_libs.BUILD b/third_party/rocm_device_libs/rocm_device_libs.BUILD
index ee162b63ab..98a55618a4 100644
--- a/third_party/rocm_device_libs/rocm_device_libs.BUILD
+++ b/third_party/rocm_device_libs/rocm_device_libs.BUILD
@@ -28,36 +28,12 @@ cc_binary(
     ],
 )
 
-bitcode_library(
+alias(
     name = "ocml",
-    srcs = glob([
-        "ocml/src/*.cl",
-    ]),
-    hdrs = glob([
-        "ocml/src/*.h",
-        "ocml/inc/*.h",
-        "irif/inc/*.h",
-        "oclc/inc/*.h",
-    ]),
-    file_specific_flags = {
-        "native_logF.cl": ["-fapprox-func"],
-        "native_expF.cl": ["-fapprox-func"],
-        "sqrtF.cl": ["-cl-fp32-correctly-rounded-divide-sqrt"],
-    },
+    actual = "@local_config_rocm//rocm:ocml"
 )
 
-bitcode_library(
+alias(
     name = "ockl",
-    srcs = glob([
-        "ockl/src/*.cl",
-        "ockl/src/*.ll",
-    ]),
-    hdrs = glob([
-        "ockl/inc/*.h",
-        "irif/inc/*.h",
-        "oclc/inc/*.h",
-    ]),
-    file_specific_flags = {
-        "gaaf.cl": ["-munsafe-fp-atomics"],
-    },
+    actual = "@local_config_rocm//rocm:ocml"
 )
